Shared/Samples/Show line of sight between geoelements/ShowLineOfSightBetweenGeoelementsView.swift

```diff
@@ -0,0 +1,358 @@
+                    in: heightRange,
+                    step: 1
+                ) {
+                    Text("Observer Height")
+                } minimumValueLabel: {
+                    Text(heightRange.lowerBound, format: numberFormat)
+                } maximumValueLabel: {
+                    Text(heightRange.upperBound, format: numberFormat)
+                }
+                .listRowSeparator(.hidden, edges: .top)
+            }
+            .presentationDetents([.fraction(0.25)])
+            .navigationTitle("Settings")
+            .navigationBarTitleDisplayMode(.inline)
+            .toolbar {
+                ToolbarItem(placement: .confirmationAction) {
+                    Button("Done") {
+                        isPresented = false
+                    }
+                }
+            }
+        }
+    }
+}
+
+private extension ShowLineOfSightBetweenGeoelementsView {
+    /// View model responsible for setting up the 3D scene, graphics, animation,
+    /// and performing line of sight analysis between moving and static geoelements.
+    @MainActor
+    @Observable
+    final class Model {
+        /// Set of predefined waypoints for animating the taxi's movement.
+        private let points = [
+            Point(x: -73.984513, y: 40.748469, spatialReference: .wgs84),
+            Point(x: -73.985068, y: 40.747786, spatialReference: .wgs84),
+            Point(x: -73.983452, y: 40.747091, spatialReference: .wgs84),
+            Point(x: -73.982961, y: 40.747762, spatialReference: .wgs84)
+        ]
+        
+        /// Height of the observer in meters. Updates the observer graphic when changed.
+        var height = 20.0 {
+            didSet {
+                changeObserverHeight(height)
+            }
+        }
+        
+        private var frameIndex = 0
+        private let frameMax = 120
+        private var pointIndex = 0
+        
+        var error: Error?
+        
+        /// The 3D scene containing basemap, elevation, and building layers.
+        let scene: ArcGIS.Scene = {
+            // Creates a scene and set an initial viewpoint.
+            let scene = Scene(basemapStyle: .arcGISImagery)
+            // Add base surface from elevation service.
+            let elevationSource = ArcGISTiledElevationSource(url: .elevationService)
+            let surface = Surface()
+            surface.addElevationSource(elevationSource)
+            scene.baseSurface = surface
+            var buildingLayer = ArcGISSceneLayer(url: .buildingsService)
+            scene.addOperationalLayer(buildingLayer)
+            return scene
+        }()
+        
+        /// Graphics overlay used to render the observer and target symbols.
+        var graphicsOverlay = GraphicsOverlay() {
+            didSet {
+                let renderer = SimpleRenderer()
+                renderer.sceneProperties.headingExpression = ("[HEADING]")
+                graphicsOverlay.renderer = renderer
+            }
+        }
```

Comment:
```suggestion
        let graphicsOverlay: GraphicsOverlay = {
            let overlay = GraphicsOverlay()
            let renderer = SimpleRenderer()
            renderer.sceneProperties.headingExpression = ("[HEADING]")
            overlay.renderer = renderer
            return overlay
        }()
```
I prefer closure-based initialization over `didSet` as `didSet` is a 2-stage process. Also `let` instead of `var` when possible.